compiler/src/dotty/tools/dotc/core/classfile/ClassfileParser.scala

```diff
@@ -1127,7 +1127,11 @@ class ClassfileParser(
       val outerName = entry.strippedOuter
       val innerName = entry.originalName
       val owner = classNameToSymbol(outerName)
-      val result = atPhase(typerPhase)(getMember(owner, innerName.toTypeName))
+      val result = owner.denot.infoOrCompleter match
+        case _: StubInfo if hasAnnotation(entry.jflags) =>
+          requiredClass(innerName.toTypeName)
```

Comment:
Since this is not obvious, could you add a comment explaining why this is needed and pointing the reader to the test case?